taskvine/src/manager/vine_manager.c

```diff
@@ -3665,45 +3697,55 @@ static int send_one_task(struct vine_manager *q, int *tasks_ready_left_to_consid
 		struct vine_worker_info *w = vine_schedule_task_to_worker(q, t);
 		q->stats->time_scheduling += timestamp_get() - q->stats_measure->time_scheduling;
 
-		if (w) {
-			priority_queue_remove(q->ready_tasks, t_idx);
+		/* task is runnable but no worker is fit, demote it and consider later */
```

Comment:
I don't think we want this as it impacts greatly priority. When resources become available, tasks with lower priority may be scheduled first because higher priority tasks are in the blocked list.